### PR TITLE
LPS-183807 We need to create DDMFormFieldOptions from Data Provider before evaluating the DDMForm

### PR DESCRIPTION
**Pull request by marcelabc**
Related issue: https://issues.liferay.com/browse/LPS-183807

### CODE REVIEW

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/portlet/action/AddFormInstanceRecordMVCActionCommand.java [hunk 1]
```diff
@@ -202,6 +211,29 @@ protected DDMForm getDDMForm(DDMFormInstance ddmFormInstance)
...
+			if (Objects.equals(ddmFormField.getType(), "select") &&
+				Objects.equals(
+					ddmFormField.getDataSourceType(), "data-provider")) {
+
+				ddmFormField.setProperty(
+					"options",
+					_ddmFormFieldOptionsFactory.create(
+						ddmFormField, ddmFormFieldRenderingContext));
+			}
+		}
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/portlet/action/AddFormInstanceRecordMVCActionCommand.java:234]: @marcelabc do we also need to loop through `ddmFormField.getNestedDDMFormFields()` here??

Comment (marcelabc) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/portlet/action/AddFormInstanceRecordMVCActionCommand.java]: Thanks!

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
ci:forward

**Pull request closed without merging**